changelog/issue-7628.md

```diff
@@ -0,0 +1,15 @@
+audience: worker-deployers
+level: minor
+reference: issue 7628
+---
+Generic Worker: adds a `Metadata` task feature (controlled with `config.enableMetadata` [default: `true`]) that writes out a file `generic-worker-metadata.json` (in the current working directory of the generic worker process) containing information about the last task run.
+
+Currently, the file will look something like this:
+
+```json
+{
+	"lastTaskUrl": "https://firefox-ci-tc.services.mozilla.com/tasks/Klc17PU-TMmo4axlfihKJQ/runs/0"
+}
+```
+
+Additional data may be added to this file in future releases.
```

Comment:
Let's explicitly state that the feature cannot be enabled/disabled in the task, as that might be confusing (it confused me at first). Alternatively the changelog doesn't need to mention that it is a task feature, it could just call it a feature, because normally task features relate to what is inside `task.payload.features` and are enabled/disabled with a boolean there, which is not the case here.

Reply:
Addressed in the first paragraph of the changelog (with a force commit):

>Generic Worker: adds a `Metadata` feature (controlled with worker config `enableMetadata` [default: `true`], not controllable in the task payload) that writes out a file `generic-worker-metadata.json` (in the current working directory of the generic worker process) containing information about the last task run.